test/src/materials/ADCoupledMaterial.C

```diff
@@ -0,0 +1,45 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+#include "ADCoupledMaterial.h"
+
+registerADMooseObject("MooseTestApp", ADCoupledMaterial);
+
+defineADValidParams(
+    ADCoupledMaterial, ADMaterial, params.addRequiredCoupledVar("coupled_var", "A coupledvariable");
+    params.addRequiredParam<MaterialPropertyName>("ad_mat_prop",
+                                                  "Name of the ad property this material defines");
+    params.addRequiredParam<MaterialPropertyName>(
+        "regular_mat_prop", "Name of the regular property this material defines"););
+
+template <ComputeStage compute_stage>
+ADCoupledMaterial<compute_stage>::ADCoupledMaterial(const InputParameters & parameters)
+  : ADMaterial<compute_stage>(parameters),
+    _ad_mat_prop(this->template declareADProperty<Real>(
+        this->template getParam<MaterialPropertyName>("ad_mat_prop"))),
+    _regular_mat_prop(this->template declareProperty<Real>(
+        this->template getParam<MaterialPropertyName>("regular_mat_prop"))),
+    _coupled_var(this->template adCoupledValue<compute_stage>("coupled_var"))
+{
+}
+
+template <ComputeStage compute_stage>
+void
+ADCoupledMaterial<compute_stage>::computeQpProperties()
+{
+  _regular_mat_prop[_qp] = 4.0 * _coupled_var[_qp].value();
+  _ad_mat_prop[_qp] = 4.0 * _coupled_var[_qp];
+}
+
+template <>
+void
+ADCoupledMaterial<RESIDUAL>::computeQpProperties()
```

Comment:
What's the situation with this?  Is it needed... or are you just showing that it's possible (to have a specialization for RESIDUAL)?

Hopefully it's not needed!

Reply:
This is specific to this test...it's demonstrating that you get bad convergence if you drop the derivative information from the couple variable

Reply:
Nice!  Ok - so could you add a comment here.  I could see people spotting this and thinking it's the _right_ way to do things!